File: leetcode/283. Move Zeros/283_Move_zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nums_size = nums.size();
        for(int i=0; i<nums_size; i++){
            if (nums[i] == 0){
                for(int j=i+1; j <nums_size; j++){
                    if(nums[j] != 0){
                        int temp;
                        temp = nums[i];
                        nums[i] = nums[j];
                        nums[j] = temp;
                        break;
                    }
                }
            }
        }
    }
};

File: leetcode/283. Move Zeros/283_Move_Zeros.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int none_zero = 0;
        for(int i=0; i<nums.Length; i++)
        {
            if(nums[i] != 0)
            {
                int temp = nums[i];
                nums[i] = nums[none_zero];
                nums[none_zero] = temp;
                none_zero += 1;
            }
        }
    }
}
